receiver/dcgmreceiver/util.go

```diff
@@ -19,39 +19,99 @@ package dcgmreceiver
 
 import (
 	"fmt"
-	"time"
 
 	"github.com/NVIDIA/go-dcgm/pkg/dcgm"
 )
 
-var nowUnixMicro = func() int64 { return time.Now().UnixNano() / 1e3 }
-
-// rateIntegrator converts timestamped values that represent rates into
-// cumulative values. It assumes the rate stays constant since the last
-// timestamp.
-type rateIntegrator[V int64 | float64] struct {
-	lastTimestamp    int64
-	aggregatedRateUs V // the integration of the rate over microsecond timestamps.
+// For each metric, we need to track:
+type metricStats struct {
+	// Timestamp (us)
```

Comment:
You used unicode in other comments…
```suggestion
	// Timestamp (µs)
```